libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3261,8 +3288,12 @@ void TypeChecker::requireLValue(Expression const& _expression, bool _ordinaryAss
 				if (varDecl->isExternalCallableParameter() && dynamic_cast<ReferenceType const*>(identifier->annotation().type))
 					return { 7128_error, "External function arguments of reference type are read-only." };
 
+		fatal = true;
 		return { 4247_error, "Expression has to be an lvalue." };
```

Comment:
This is the "fallback error message" - is it possible to create a more detailed error message for some of the new failure cases here?